### PR TITLE
feat: split KMM into KMM and uKMM (#37)

### PR DESCRIPTION
**Pull request by xukai92**
This PR addresses https://github.com/JuliaML/DensityRatioEstimation.jl/issues/37 by introducing `AbstractKMM` and subtyping `KMM` and `uKMM`. Shared functions are defined over `AbstractKMM` to avoid repeated codes.

### CODE REVIEW

File: src/kmm/julia.jl [hunk 1]
```diff
@@ -2,20 +2,7 @@
 # Licensed under the MIT License. See LICENSE in the project root.
 # ------------------------------------------------------------------
 
-# NOTE: this function is a hack for Zygote.jl compatbility; see lib/zygote.jl
-function warn_kmm_julialib(B, ϵ)
```
Comment (xukai92) [inline src/kmm/julia.jl:6]: This function is removed as `uKMM` by construction doesn't have these two fields.

**Comment by juliohm** [comment]
I triggered the v0.6 minor release. Let me know what do you think about a major v1.0 :)

**Comment by xukai92** [comment]
A major v1.0 sounds good to me. 
Should somehow mention that there are both `KMM` and `uKMM` in the README.md?
Not sure what's the best way to do so in the current presentation.

**Comment by juliohm** [comment]
That is a good point. Would you like to give it a try in the README? I'd be happy to review the exposition.

**Comment by xukai92** [comment]
I accidentally pushed to master directly. What's the best way to fix it?

**Comment by xukai92** [comment]
OK I fixed it by removing the commit. I will start a PR for the README.md updating.

**Pull request merged**